External/FreeSASA/RDFreeSASA.cpp

```diff
@@ -0,0 +1,209 @@
+//  Copyright (c) 2016, Novartis Institutes for BioMedical Research Inc.
+//  All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+//       copyright notice, this list of conditions and the following
+//       disclaimer in the documentation and/or other materials provided
+//       with the distribution.
+//     * Neither the name of Novartis Institutes for BioMedical Research Inc.
+//       nor the names of its contributors may be used to endorse or promote
+//       products derived from this software without specific prior written
+//       permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/QueryOps.h>
+#include <GraphMol/RDKitQueries.h>
+#include <GraphMol/MonomerInfo.h>
+#include <RDGeneral/types.h>
+#include "RDFreeSASA.h"
+#include "boost/format.hpp"
+
+extern "C" {
+#include "freesasa.h"
+}
+
+namespace RDKit {
+  namespace common_properties {
+    namespace Atom {
+      const std::string SASA = "SASA";; // Solvent Accessible Surface Area for atom- double
+      const std::string SASAClass = "SASAClass";     // Class type, 0,1,2... etc
+      const std::string SASAClassName = "SASAClassName"; // Class name, Polar, APolar etc...
+    }
+    namespace Molecule {
+      const std::string SASA = "SASA"; // Total Solvent Accessible Surface area for molecule;
+    }
+  }
+}
+
+namespace FreeSASA {
+using namespace RDKit;
+
+bool classifyAtoms(ROMol &mol, std::vector<double> &radii,
```

Comment:
Because of the way this is set up, on output `radii` only contains information about atoms that had monomer information. There's not really any effective way to map that to real atom indices.
As long as the code only works with monomer info, `radii` should either be an `std::map` from atom index to radius *or* the function should fail when it encounters an atom without monomer info.